src/OauthTokenFileStorage.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge;
+
+use Apigee\Edge\HttpClient\Plugin\Authentication\OauthTokenStorageInterface;
+use Drupal\apigee_edge\Form\AuthenticationForm;
+
+/**
+ * Storing and returning OAuth access token data.
+ */
+class OauthTokenFileStorage implements OauthTokenStorageInterface {
+
+  /**
+   * Ensures that token gets refreshed earlier than it expires.
+   *
+   * Number of seconds extracted from token's expiration date when
+   * hasExpired() calculates.
+   *
+   * @var int
+   */
+  protected $leeway = 30;
+
+  /**
+   * An internally cached token data store.
+   *
+   * @var array
+   */
+  private static $token_data;
```

Comment:
This is a service, therefore it is like a singleton (unless it is a shared service) in a page request. So you do not need a static variable as a cache, non static variables also behaves as cache and stores the information until the object has been destructed (page request is active).

Reply:
LGTM